src/entrypoints/is-array.d.ts

```diff
@@ -1,3 +1,3 @@
 interface ArrayConstructor {
-  isArray(arg: any): arg is unknown[];
+  isArray<T>(arg: 0 extends 1 & T ? never : T): arg is 0 extends 1 & T ? never : T extends unknown[] | readonly unknown[] ? T : T[] extends T ? T[] : never;
```

Comment:
Not sure if you saw my comment [here](https://github.com/total-typescript/ts-reset/pull/56#issuecomment-1595405064) -- you can simplify this by removing the check for `unknown[]` entirely.

```suggestion
  isArray<T>(arg: 0 extends 1 & T ? never : T): arg is 0 extends 1 & T ? never : T extends readonly unknown[] ? T : T[] extends T ? T[] : never;
```

Reply:
@ahrjarrett Thanks, I saw your comment but it slipped my mind while I was sitting with the code